pkg/build/api/v1beta3/types.go

```diff
@@ -85,6 +85,9 @@ type BuildStatus struct {
 
 	// Config is an ObjectReference to the BuildConfig this Build is based on.
 	Config *kapi.ObjectReference `json:"config,omitempty"`
+
+	// Revision is the revision that was built with this build
+	Revision *SourceRevision `json:"revision,omitempty" description:"source revision built with this build"`
 }
```

Comment:
> Why aren't we setting the spec field and making it immutable once set?

Because I thought of this more a status, but I don't have an objection to using the spec field. 




> Why do we need a new API endpoint? Why is the API endpoint different?

Because we want to give the builder service account only the access it needs.